DigitalLearningSolutions.Data/Models/User/UserAccountSet.cs

```diff
@@ -5,34 +5,33 @@
 
     public class UserAccountSet
     {
-        public readonly AdminUser? AdminAccount;
+        public readonly List<AdminUser> AdminAccounts;
         public readonly List<DelegateUser> DelegateAccounts;
 
         public UserAccountSet() : this(null, null) { }
 
-        public UserAccountSet(AdminUser? adminAccount, IEnumerable<DelegateUser?>? delegateAccounts)
+        public UserAccountSet(List<AdminUser>? adminAccounts, IEnumerable<DelegateUser?>? delegateAccounts)

```

Comment:
I fancy you can split this in two.
Conceptually, the `UserAccountSet`, when structured as `(AdminUser?, List<DelegateUser>)`, was very different from `(List<AdminUser>, List<DelegateUser>)`, because it represented the set of accounts belonging to a particular user. So that's how it's used in ChangePasswordController (which also uses `Any()` and `GetUserRefs()`).

However, it's being used in the Login procedure as a wrapper around a tuple basically, because it's really working with the List<AU> and List<DU> separately and not using the methods. Perhaps it would make sense for the return value of AttemptLogin to contain a `UserAccountSet`, because at that point you do have a set of accounts belonging to a particular person.

The uses in the Login procedure are disjoint from the uses in the ChangePassword procedure, I think.

So I think it would be better to create a new type, something like `AccountCollection`, or even simply `AccountSet` (though that does imply having sets rather than lists), for the login procedure, then leave `UserAccountSet` as it was for the ChangePassword procedure to use.

Reply:
Reverted all the changes here back to AdminUser?. Kevin is OK with us throwing if we manage to get 2 unverified admins at step one, so I've reverted that step to take a first or default, which means all the Verify methods go back to taking a single admin.